File: Java/src/oop/CoffeeEx.java
package oop;

import java.util.List;

class Customer {
	// menuItem과 barista에 대한 객체 접근
	public void order(String menuName, Menu menu, Barista barista) {
		MenuItem menuItem = menu.chooseCoffee(menuName);
		Coffee coffee = barista.makeCoffee(menuItem);
	}
}
class MenuItem{
	private String name;
	private int price;
	public MenuItem(String name, int price) {
		this.name = name;
		this.price = price;
	}
	public String getName() {
		return name;
	}
	public int getPrice() {
		return price;
	}

	
}
class Menu{
	private List<MenuItem> items;
	public Menu(List<MenuItem> items) {
		this.items = items;
	}
	
	// 커피 고르기
	public MenuItem chooseCoffee(String name) {
		for(MenuItem item : items) {
			if(item.getName().equals(name)){
				return item;
			}
		}
		return null;
	}
	
}
class Barista {
	public Coffee makeCoffee(MenuItem menuitem) {
		Coffee coffee = new Coffee(menuitem);
		return coffee;
	}
	
}
class Coffee {
	private String name;
	private int price;
	public Coffee(MenuItem menuItem) {
		name = menuItem.getName();
		price = menuItem.getPrice();
	}
}


public class CoffeeEx {

	public static void main(String[] args) {
		// TODO Auto-generated method stub

	}

}

File: Python/CoffeeEx.py
#!/usr/bin/env python
# coding: utf-8


class Customer:
    def order(menuName, menu, barista):
        menuItem = menu.chooseCoffee(menuName)
        coffee= barista.makeCoffee(menuItem)
class MenuItem:
    def __init__(self, name,price):
        self.__name = name
        self.__price = price
    def getName(self):
        return sef.__name
    def getPrice(self):
        return sef.__price
class Menu:
    def __init__(self, items):
        this.items = items
    def choooseCoffee(name):
        for item in items:
            if(item.name == name):
                return item
        return null
class Barista:
    def makeCoffee(menuitem):
        coffee = Coffee(menuitem)
        return coffee
class Coffee:
    def __init__(self, menuItem):
        self.__name = menuItem.name
        self.__price = menuItem.price
    




